soc/atmel/sam/Kconfig

```diff
@@ -0,0 +1,7 @@
+# Copyright (c) 2024 Gerson Fernando Budke <nandojve@gmail.com>
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_FAMILY_SAM
```

Comment:
This Kconfig needs to match the name, the name is "atmel_sam" so needs to be `SOC_FAMILY_ATMEL_SAM`

Reply:
Hi @nordicjm ,

This seems to be inconsistent in the tree then. This is one example https://github.com/zephyrproject-rtos/zephyr/blob/6acabd159aeeb644b8dc9d17012e14d25ce9684d/soc/st/stm32/Kconfig.soc#L6

We are using the same structure vendor/family. If I change this I need to update the hal for two different families SAM/SAM0.

Can we clarify this ?

Reply:
I was wondering if it is not possible perform this change after the collab branch be merged in the main. Why? Then it is easy to perform the tree wide change since it requires hal change too. We can open an issue to address as a suggestion.

Reply:
Hi @nordicjm ,

Did you had time to check this?

Reply:
it would be nice to get it aligned with: https://github.com/zephyrproject-rtos/zephyr/blob/c2d5d076c5d901188384cdfc5cbe5cfe3e292663/soc/atmel/sam/soc.yml#L2

especially as we plan for having compliance checks in place in future to ensure such names are aligned between Kconfig, CMake, yml-files in order to avoid the legacy mess of names being different across tools.

Could you place a followup comment https://github.com/zephyrproject-rtos/zephyr/issues/51831 under "Open issues" and place your name next to it 😄 

Then I'd be happy to accept this part of PR as-is.

You are of course also welcome to create a dedicated issue for tracking such work.

Reply:
Added entry at #51831 and created the issue #69046 to easy track.